open-craft/frontend-app-course-authoring
# feat: copy & paste units

**Description:**

Adds copy and paste units functionality as described [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3866460192/New+Additions+to+Course+Content+Outline#Copy-%26-Paste).

**Test instructions:**

* Get devstack up and running using `make {lms,cms}-up`.
* Start npm devserver in this repo using `npm start`.
* Enable copy paste functionality by adding `contentstore.enable_copy_paste_units` & `new_studio_mfe.use_tagging_taxonomy_list_page` waffle flags in [admin](http://localhost:18010/admin/waffle/flag/)
* You might need to run `make {cms,lms}-migrate` if the API calls start to fail after enabling above waffle flags.
* Goto http://localhost:2001/course/course-v1:edX+DemoX+Demo_Course/
* Make sure paste button is not visible.
* Copy any unit from its action menu.
* Paste button should be visible along with a small label below it, on which you can hover to see copied unit details and you can click on it to open the unit in a new tab.
* Broadcast channel is used to make sure clipboard is synchronized across browser tabs.
